File: backend/User_Management/findUser/routes/findRoutes.rb
require 'sinatra'
require 'json'
require_relative '../model/user'

get '/user/:id' do
  content_type :json
  user = User.find_by(id: params[:id])
  if user
    user.to_json(only: [:id, :username, :email, :role])
  else
    status 404
    { error: 'User not found' }.to_json
  end
end

get '/username/:name' do
  content_type :json
  users = User.where("username LIKE ?", "%#{params[:name]}%")
  users.to_json(only: [:id, :username, :email, :role])
end

File: backend/User_Management/findUser/routes/findRoutes.py
from flask import Blueprint, request, jsonify
from model.findModel import User

find_bp = Blueprint('find_bp', __name__)

@find_bp.route('/find/username/<string:username>', methods=['GET'])
def find_by_username(username):
    """
    Find users by partial username.
    ---
    parameters:
      - name: username
        in: path
        type: string
        required: true
        description: Partial or full username to search for.
    responses:
      200:
        description: List of users matching the username.
        schema:
          type: array
          items:
            type: object
            properties:
              id:
                type: integer
                example: 1
              username:
                type: string
                example: johndoe
              email:
                type: string
                example: johndoe@email.com
              role:
                type: string
                example: admin
      404:
        description: No users found.
        schema:
          type: object
          properties:
            error:
              type: string
              example: No users found
    examples:
      application/json: |
        GET /find/username/john
    """
    users = User.query.filter(User.username.like(f"%{username}%")).all()
    if not users:
        return jsonify({'error': 'No users found'}), 404
    return jsonify([
        {
            'id': user.id,
            'username': user.username,
            'email': user.email,
            'role': user.role
        } for user in users
    ])
